main.py

```diff
@@ -261,14 +261,24 @@ def _kill_tb_child():
     atexit.register(_kill_tb_child)
 
 
-def get_best_checkpoint_path(run_dir, phase):
+def get_best_checkpoint_path(run_dir, phase, task_name=None):
     """ Look in run_dir for model checkpoint to load.
     Hierarchy is
-        1) best checkpoint from the phase so far
-        2) if we do only target training without pretraining, then load checkpoint before 
+        1) best task-specific checkpoint
+        2) best checkpoint from the phase so far
+        3) if we do only target training without pretraining, then load checkpoint before 
         target training
-        3) nothing found (empty string) """
-    checkpoint = glob.glob(os.path.join(run_dir, "model_state_%s_epoch_*.best_macro.th" % phase))
+        4) nothing found (empty string) 
+    """
+    checkpoint = []
+    if task_name is not None:
```

Comment:
This doesn't seem like the right test. Even if it works for now, it'd be easy to break in future editing.
- In pretraining, we never want to do this, since there's only ever one task, and it's helpful to have a common filename standard for pretrained models.
- In target task training, we always want to do this, even if there's only one target task, since it'll make our checkpoints robust in case we come back and add more target task laters.

Let's make this depend on the phase.

Reply:
What do you mean that there is only one task in pretraining? 
Also, we only pass in task_name during target_train phase (and we don't' during pretrain), but I can move that logic inside this function.

Reply:
Er, sorry. There's only ever one model and set of checkpoints in pretraining, regardless of how many tasks we use.

I know the presence of the task variable depends on phase, but it's easier to maintain if we directly check the phase in this if statement.